core/src/main/java/org/elasticsearch/gateway/GatewayService.java

```diff
@@ -202,7 +202,7 @@ private void performStateRecovery(boolean enforceRecoverAfterTime, String reason
                 threadPool.schedule(recoverAfterTime, ThreadPool.Names.GENERIC, () -> {
                     if (recovered.compareAndSet(false, true)) {
                         logger.info("recover_after_time [{}] elapsed. performing state recovery...", recoverAfterTime);
-                        gateway.performStateRecovery(recoveryListener);
+                        gateway.performStateRecovery(recoveryListener, Math.max(1, recoverAfterMasterNodes));
```

Comment:
higher up in this file we refer to min master nodes explicitly with a vague todo which you missed because it's not done via a code link. I think it's OK to leave the hard coded string for now (as we plan to move this things to another place anyway), but I think it's good to change the comment to explain that you can't use the settings object because it's a module and that there are plans to fix it.